java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c

```diff
@@ -615,3 +615,22 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addROC
     throwOrtException(jniEnv,convertErrorCode(ORT_INVALID_ARGUMENT),"This binary was not compiled with ROCM support.");
   #endif
 }
+
+/*
+ * Class::    ai_onnxruntime_OrtSession_SessionOptions
+ * Method:    addXnnpack
+ * Signature: (JILjava/lang/String)V
+ */
+JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addXnnpack(
+    JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong optionsHandle) {
+  (void)jobj;
+#ifdef USE_XNNPACK
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtSessionOptions* options = (OrtSessionOptions*)optionsHandle;
+  checkOrtStatus(jniEnv, api, api->SessionOptionsAppendExecutionProvider(options, "XNNPACK", 0, 0, 0));
```

Comment:
Aren't the first two zero arguments here pointers? In which case they should be `NULL`. I think it would be better to directly expose a function on `SessionOptions` `void addXnnpack(Map<String,String>)` that unpacks the map into two string arrays and passes them into the native code.

Reply:
Yeah, maybe it's better to pass config-pairs by a new function signature `void addXnnpack(Map<String,String>)`.

I was thinking if it's worth to do that since there is no extra options or arguments required by XnnpackEp at this moment.

Will do it tomorrow in case it's used in the future.

Reply:
Ah ok. If there aren't any options at the moment, then it's probably worth documenting that,  or putting a link to where such options would be documented in the javadoc for the function.

Reply:
Supportted `ProviderOptions` already, new document is required once added new options for XNNPACK EP